x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -171,84 +163,109 @@ export async function installTemplateForDataStream({
   });
 }
 
+interface TemplateMapEntry {
+  template:
+    | {
+        mappings: NonNullable<RegistryElasticsearch['index_template.mappings']>;
+      }
+    | {
+        settings: NonNullable<RegistryElasticsearch['index_template.settings']> | object;
+      };
+}
+type TemplateMap = Record<string, TemplateMapEntry>;
 function putComponentTemplate(
-  body: object | undefined,
-  name: string,
-  esClient: ElasticsearchClient
-): { clusterPromise: Promise<any>; name: string } | undefined {
-  if (body) {
-    const esClientParams = {
-      name,
-      body,
-    };
-
-    return {
-      // @ts-expect-error body expected to be ClusterPutComponentTemplateRequest
-      clusterPromise: esClient.cluster.putComponentTemplate(esClientParams, { ignore: [404] }),
-      name,
-    };
+  esClient: ElasticsearchClient,
+  params: {
+    body: TemplateMapEntry;
+    name: string;
+    create?: boolean;
   }
+): { clusterPromise: Promise<any>; name: string } {
+  const { name, body, create = false } = params;
+  return {
+    clusterPromise: esClient.cluster.putComponentTemplate(
+      // @ts-expect-error body is missing required key `settings`. TemplateMapEntry has settings *or* mappings
+      { name, body, create },
+      { ignore: [404] }
+    ),
+    name,
+  };
 }
 
-function buildComponentTemplates(registryElasticsearch: RegistryElasticsearch | undefined) {
-  let mappingsTemplate;
-  let settingsTemplate;
+const mappingsSuffix = '-mappings';
+const settingsSuffix = '-settings';
+const userSettingsSuffix = '-user_settings';
+type TemplateBaseName = string;
+type UserSettingsTemplateName = `${TemplateBaseName}${typeof userSettingsSuffix}`;
+
+const isUserSettingsTemplate = (name: string): name is UserSettingsTemplateName =>
+  name.endsWith(userSettingsSuffix);
+
+function buildComponentTemplates(
```

Comment:
similar to what we do for the base index template, can we attach meta information to all these component templates?
https://github.com/elastic/kibana/blob/a816dae16249b1c86a0e1f719d63d3f047192c79/x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts#L369

the component templates meta should just contain package information:
```ts
  const _meta = {
    package: {
      name: packageName,
    },
  };
```